contracts/math/decimal.ral

```diff
@@ -1,10 +1,38 @@
+struct SqrtPrice{
+    mut v: U256
+}
+
+struct Liquidity{
+    mut v: U256
+}
+
+struct FeeGrowth{
+    mut v: U256
+}
+
+struct Percentage{
+    mut v: U256
+}
+
+struct TokenAmount{
+    mut v: U256
+}
+
+struct FixedPoint{
+    mut v: U256
+}
+
 Abstract Contract Decimal() {
     const SqrtPriceScale = 24
     const LiquidityScale = 5
     const FeeGrowthScale = 28
     const FixedPointScale = 12
     const PercentageScale = 12
     const TokenAmountScale = 0
+    const SqrtPriceDenominator = 1000000000000000000000000
+    const LiquidiyDenominator = 100000
+    const FeeGrowthDenominator = 10000000000000000000000000000
+    const PercentageDenominator = 1000000000000
```

Comment:
I think we shouldn't keep hard coded denominators in ralph. If we introduce them we will always need to check if the type scale match the denominator variable since they are in relationship. It would be the best if ralph compilator would allow `const SqrtPriceDenominator = one(SqrtPriceScale)` syntax, but as far as I know it is not possible. I think it is better to keep only their scales in ralph and use `one()` function.

Maybe we can also add `types.ral` file which will store all the structs with their scales. It can come within this PR or after the dependency upgrade to store global constants. It can also work for `errors.ral` `swap_structs.ral` etc..

Reply:
Must have misunderstood https://github.com/invariant-labs/protocol-alephium/pull/115#discussion_r1691119859, reverting